packages/openbox.rb

```diff
@@ -0,0 +1,57 @@
+require 'package'
+
+class Openbox < Package                 # The first character of the class name must be upper case
+  description 'Openbox is a highly configurable, next generation window manager with extensive standards support.'
+  homepage 'http://openbox.org/'
+  version '3.6.1'
+  compatibility 'all'
+  source_url 'http://openbox.org/dist/openbox/openbox-3.6.1.tar.xz'
+  source_sha256 'abe75855cc5616554ffd47134ad15291fe37ebbebf1a80b69cbde9d670f0e26d'   # Use the command "sha256sum"
+
+  depends_on 'libev'
+  depends_on 'startup_notification'
+  depends_on 'xcb_util_cursor'
+  depends_on 'xcb_util_keysyms'
+  depends_on 'xcb_util_wm'
+  depends_on 'xcb_util_xrm'
+  depends_on 'yajl'
+  depends_on 'wayland' => :build
+  depends_on 'xorg_proto'
+  depends_on 'glib'
+  depends_on 'libxfixes'
+  depends_on 'libsm'
+  depends_on 'pango' => :build
+  depends_on 'libx11' => :build
+  depends_on 'adwaita_icon_theme'
+  depends_on 'sommelier'
+
+  ENV['CFLAGS'] = "-lX11 -lXau"
+  def self.build                   # the steps required to build the package
+    system "./configure #{CREW_OPTIONS}"
+    system "make"
+  end
+
+  def self.install                 # the steps required to install the package
+    system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
+    system "echo '#!/bin/bash' > startopenbox"
+    system "echo 'stopsommelier' >> startopenbox"
+    system "echo 'echo am start x.org.server/.MainActivity | sudo android-sh' >> startopenbox"
+    system "echo 'echo Starting XServer XSDL' >> startopenbox"
+    system "echo 'sleep 10' >> startopenbox"
```

Comment:
10 seconds seems like too long to wait.

Reply:
It takes 7 seconds on my phone (is my phone too old lol)
I think 7 seconds is enough for most chromebook

Reply:
You could also poll for xserver availability after exporting the display and before running openbox in the script, using xdpyinfo like here:  https://unix.stackexchange.com/a/418903

chromebrew doesn't have xdpyinfo, but importing a recipe for that from arch took about 5 min:

```
# Adapted from Arch Linux xorg-xdpyinfo PKGBUILD at:
# https://github.com/archlinux/svntogit-packages/raw/packages/xorg-xdpyinfo/trunk/PKGBUILD

require 'package'

class Xorg_xdpyinfo < Package
  description 'Display information utility for X'
  version '1.3.2-4'
  compatibility 'all'
  source_url 'https://xorg.freedesktop.org/archive/individual/app/xdpyinfo-1.3.2.tar.bz2'
  source_sha256 '30238ed915619e06ceb41721e5f747d67320555cc38d459e954839c189ccaf51'

  depends_on 'libx11'
  depends_on 'libxext'
  depends_on 'libxtst'
  depends_on 'libxxf86vm'
  depends_on 'libxrender'
  depends_on 'libxcomposite'
  depends_on 'libxinerama'
  depends_on 'xorg_proto' 

def self.build
system "./configure #{CREW_OPTIONS}"
system 'make'
end
def self.install
system "make DESTDIR=#{CREW_DEST_DIR} install"
end
end
```

Reply:
Should we add this as a new package @satmandu?

Reply:
Yes, go ahead.

Just for completeness' sake, here is roughly what I would assume would be the wait code for starting openbox:

```
system "echo 'echo Starting XServer XSDL' >> startopenbox"
system "export DISPLAY=100.115.92.2:0' >> startopenbox
system "cat <EOF>> startopenbox
# Wait for XSDL X server
MAX_ATTEMPTS=120 # About 60 seconds
COUNT=0
echo -n \"Waiting for X server to be ready...\"
while ! xdpyinfo -display ${DISPLAY} >/dev/null 2>&1; do
  echo -n \".\"
  sleep 0.50s
  COUNT=$(( COUNT + 1 ))
  if [ \"${COUNT}\" -ge \"${MAX_ATTEMPTS}\" ]; then
    echo \"  Gave up waiting for X server on ${DISPLAY}\"
    exit 1
  fi
done
echo \"  Done - X server is ready!\"
EOF"
echo 'openbox' >> startopenbox
```

(Obviously include xorg_xdpyinfo in your depends if this is the way you want to go.)

Reply:
> Yes, go ahead.
> 
> Just for completeness' sake, here is roughly what I would assume would be the wait code for starting openbox:
> 
> ```
> system "echo 'echo Starting XServer XSDL' >> startopenbox"
> system "export DISPLAY=100.115.92.2:0' >> startopenbox
> system "cat <EOF>> startopenbox
> # Wait for XSDL X server
> MAX_ATTEMPTS=120 # About 60 seconds
> COUNT=0
> echo -n \"Waiting for X server to be ready...\"
> while ! xdpyinfo -display ${DISPLAY} >/dev/null 2>&1; do
>   echo -n \".\"
>   sleep 0.50s
>   COUNT=$(( COUNT + 1 ))
>   if [ \"${COUNT}\" -ge \"${MAX_ATTEMPTS}\" ]; then
>     echo \"  Gave up waiting for X server on ${DISPLAY}\"
>     exit 1
>   fi
> done
> echo \"  Done - X server is ready!\"
> EOF"
> echo 'openbox' >> startopenbox
> ```
> 
> (Obviously include xorg_xdpyinfo in your depends if this is the way you want to go.)

`<<EOF>` ?

Reply:
Sure, that may make it "easier".